File: Homework 5/C++/NameAddressId.cpp
//#include <locale>
#include "NameAddressId.h"

using namespace std;

NameAddressId::NameAddressId() {

	name = FullName();
	address = Address();
	int id = 0;

}

NameAddressId::~NameAddressId() {

}

FullName NameAddressId::getFullName() {

	FullName ret = name;
	return ret;

}

void NameAddressId::setFullName(FullName input) {

	name = input;

}

Address NameAddressId::getAddressObj() {

	Address ret = address;
	return ret;

}

void NameAddressId::setAddressObj(Address input) {

	address = input;

}

SingleName NameAddressId::getFirst() {

	SingleName ret = name.getFirst();
	return ret;

}

SingleName NameAddressId::getMiddle() {

	SingleName ret = name.getMiddle();
	return ret;

}

SingleName NameAddressId::getLast() {

	SingleName ret = name.getLast();
	return ret;

}

string NameAddressId::getFullNameSpaces() {

	string space = " ";
	string concat = "";
	AlphaString aFirst = name.getFirst().get();
	AlphaString aMiddle = name.getMiddle().get();
	AlphaString aLast = name.getLast().get();

	concat.append(aFirst.get());
	concat.append(space);
	concat.append(aMiddle.get());
	concat.append(space);
	concat.append(aLast.get());

	return concat;

}

string NameAddressId::getLastCommaFirst() {

	string commaSpace = ", ";
	string concat = "";
	AlphaString aFirst = name.getFirst().get();
	AlphaString aLast = name.getLast().get();

	concat.append(aLast.get());
	concat.append(commaSpace);
	concat.append(aFirst.get());

	return concat;

}

void NameAddressId::setFirst(SingleName input) {

	name.setFirst(input);

}

void NameAddressId::setMiddle(SingleName input) {

	name.setMiddle(input);

}

void NameAddressId::setLast(SingleName input) {

	name.setLast(input);

}

void NameAddressId::setName(SingleName inputFirst, SingleName inputMiddle,
		SingleName inputLast) {

	name.setFirst(inputFirst);
	name.setMiddle(inputMiddle);
	name.setLast(inputLast);

}

void NameAddressId::setStreet(string input) {

	address.setStreet(input);

}

void NameAddressId::setCity(SingleName input) {

	address.setCity(input);

}

void NameAddressId::setState(SingleName input) {

	address.setState(input);

}

void NameAddressId::setZip(int input) {

	address.setZip(input);

}

string NameAddressId::getStreet() {

	return address.getStreet();

}

SingleName NameAddressId::getCity() {

	return address.getCity();

}

SingleName NameAddressId::getState() {

	return address.getState();

}

int NameAddressId::getZip() {

	return address.getZip();

}

void NameAddressId::setID(int input) {

	id = input;

}

int NameAddressId::getID() {

	return id;

}

bool NameAddressId::equals(NameAddressId other) {

	if ((name.equals(other.getFullName())) &&
		(address.equals(other.getAddressObj())) &&
		(id == other.getID()))
		return true;
	else return false;

}

NameAddressId NameAddressId::constructor(NameAddressId other) {

}

NameAddressId NameAddressId::copy(NameAddressId other) {

}

File: Homework 5/Java/NameAddressID.java
/*
 * Objective: Creates object that includes a full name (first middle last or last, 
 * 		first middle),	an address (street, city, state, zip), and an ID number.
 * 		Overrides clone and equals.
 */

public class NameAddressID implements Cloneable {

	private String street, spacedName, commaName;
	private SingleName city, state, first, middle, last;
	private int zip, id;

	public NameAddressID(String street, SingleName city, SingleName state,
			int zip, SingleName first, SingleName middle, SingleName last,
			int id) {
		this.first = first;
		this.middle = middle;
		this.last = last;
		this.street = street;
		this.city = city;
		this.state = state;
		this.zip = zip;
		this.id = id;
	}

	public NameAddressID() {
	}

	// copy constructor
	NameAddressID(NameAddressID nameAddID) {
		first = nameAddID.first;
		middle = nameAddID.middle;
		last = nameAddID.last;
		street = nameAddID.street;
		city = nameAddID.city;
		state = nameAddID.state;
		zip = nameAddID.zip;
		id = nameAddID.id;
	}

	void setStreet(String street) {
		this.street = street;
	}

	String getStreet() {

		return street;
	}

	void setCity(SingleName city) {
		this.city = city;
	}

	SingleName getCity() {

		return city;
	}

	void setState(SingleName state) {
		this.state = state;
	}

	SingleName getState() {

		return state;
	}

	void setZip(int zip) {
		this.zip = zip;
	}

	int getZip() {

		return zip;
	}

	void setID(int zip) {
		this.zip = zip;
	}

	int getID() {

		return zip;
	}

	public void setFirst(SingleName first) {
		this.first = first;
	}

	public SingleName getFirst() {
		return first;
	}

	public void setMiddle(SingleName middle) {
		this.middle = middle;
	}

	public SingleName getMiddle() {
		return middle;
	}

	public void setLast(SingleName last) {
		this.last = last;
	}

	public SingleName getLast() {
		return last;
	}

	public void setAll(SingleName first, SingleName middle, SingleName last) {
		this.first = first;
		this.middle = middle;
		this.last = last;
	}

	/*
	 * returns full name formatted with spaces
	 */
	public String getFullNameSpaces() {
		spacedName = first.get() + " " + middle.get() + " " + last.get();
		return spacedName;
	}

	/*
	 * returns full name formatted with last name, first "Smith, Bob T"
	 */
	public String getLastCommaFirst() {
		commaName = last.get() + ", " + first.get() + " " + middle.get();
		return commaName;
	}

	@Override
	public NameAddressID clone() {
		NameAddressID obj;
		try {
			obj = (NameAddressID) super.clone();
		} catch (CloneNotSupportedException e) {
			throw new AssertionError(e);
		}
		return obj;
	}

	@Override
	public boolean equals(Object obj) {
		if (obj == null) {
			return false;
		}
		if (!NameAddressID.class.isAssignableFrom(obj.getClass())) {
			return false;
		}
		final NameAddressID other = (NameAddressID) obj;
		if (!this.street.equals(other.street)) {
			return false;
		}
		if (!this.city.equals(other.city)) {
			return false;
		}
		if ( !this.state.equals(other.state)) {
			return false;
		}
		if ( !this.first.equals(other.first)) {
			return false;
		}
		if (!this.middle.equals(other.middle)) {
			return false;
		}
		if (!this.last.equals(other.last)) {
			return false;
		}
		if (this.zip != other.zip) {
			return false;
		}
		if (this.id != other.id) {
			return false;
		}
		return true;
	}
}
